internal/services/datadog/datadog_monitors_resource.go

```diff
@@ -0,0 +1,493 @@
+			log.Printf("[INFO] Datadog monitor %q does not exist - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("retrieving Datadog Monitor %q (Resource Group %q): %+v", id.MonitorName, id.ResourceGroup, err)
+	}
+	d.Set("name", id.MonitorName)
+	d.Set("resource_group_name", id.ResourceGroup)
+	d.Set("location", location.NormalizeNilable(resp.Location))
+	if err := d.Set("identity", flattenMonitorIdentityProperties(resp.Identity)); err != nil {
+		return fmt.Errorf("setting `identity`: %+v", err)
+	}
+	if props := resp.Properties; props != nil {
+		if err := d.Set("datadog_organization_properties", flattenMonitorOrganizationProperties(props.DatadogOrganizationProperties, d)); err != nil {
+			return fmt.Errorf("setting `datadog_organization_properties`: %+v", err)
+		}
+		if err := d.Set("user_info", flattenMonitorUserInfo(props.UserInfo, d)); err != nil {
+			return fmt.Errorf("setting `user_info`: %+v", err)
+		}
+		d.Set("monitoring_status", props.MonitoringStatus == datadog.MonitoringStatusEnabled)
+		d.Set("liftr_resource_category", props.LiftrResourceCategory)
+		d.Set("liftr_resource_preference", props.LiftrResourcePreference)
+		d.Set("marketplace_subscription_status", props.MarketplaceSubscriptionStatus)
+	}
+	d.Set("sku", flattenMonitorResourceSku("Linked"))
+	d.Set("type", resp.Type)
+	return tags.FlattenAndSet(d, resp.Tags)
+}
+
+func resourceDatadogMonitorUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Datadog.MonitorsClient
+	ctx, cancel := timeouts.ForUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.DatadogMonitorID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	body := datadog.MonitorResourceUpdateParameters{
+		Properties: &datadog.MonitorUpdateProperties{},
+	}
+	if d.HasChange("monitoring_status") {
+		monitoringStatus := datadog.MonitoringStatusDisabled
+		if d.Get("monitoring_status").(bool) {
+			monitoringStatus = datadog.MonitoringStatusEnabled
+		}
+		body.Properties.MonitoringStatus = monitoringStatus
+	}
+	if d.HasChange("tags") {
+		body.Tags = tags.Expand(d.Get("tags").(map[string]interface{}))
+	}
+
+	if _, err := client.Update(ctx, id.ResourceGroup, id.MonitorName, &body); err != nil {
+		return fmt.Errorf("updating Datadog Monitor %q (Resource Group %q): %+v", id.MonitorName, id.ResourceGroup, err)
+	}
+	return resourceDatadogMonitorRead(d, meta)
+}
+
+func resourceDatadogMonitorDelete(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Datadog.MonitorsClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.DatadogMonitorID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	future, err := client.Delete(ctx, id.ResourceGroup, id.MonitorName)
+	if err != nil {
+		return fmt.Errorf("deleting Datadog Monitor %q (Resource Group %q): %+v", id.MonitorName, id.ResourceGroup, err)
```

Comment:
We can simplify this to return fmt.Errorf("deleting of %s: %+v", id, err), because the necessary information are included in id.

Reply:
Incorporating in next commit.